### PR TITLE
Add a usage page to the documentation

### PR DESCRIPTION
**Pull request by DanielNoord**
Let's see how this works.

Edit: Left to do:
- Make sure the documentation builds within the `pre-commit`. Current challenge is forcing `pre-commit` to install `-e .` before running the `docs` makefile. That is necessary to build the most recent version of the documentation.

### CODE REVIEW

File: docs/_ext/link_fixer.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    # Set priority very low so that we run after everything else has been done
+    default_priority = 1000
+
+    def apply(self, **_: Dict[str, Any]) -> None:
+        """Apply the transformation."""
+        for node in self.document.traverse(nodes.reference):
+            if (
+                "refuri" in node
+                and node["refuri"].startswith(
+                    "https://pydocstringformatter.readthedocs.io"
```
Comment (Pierre-Sassoulas) [inline docs/_ext/link_fixer.py:25]: This should be a constant imo. Well we can search and replace it in this little script, so it's not that important.

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM

**Comment by DanielNoord** [comment]
**Needs to be rebased due to commit order**

@Pierre-Sassoulas I made it a variable.

See https://github.com/DanielNoord/pydocstringformatter/runs/5253162943?check_suite_focus=true for what the CI check is doing.
Basically, if something has been changed that should be reflected in the docs and the docs haven't been rebuilt locally this will warn us.
At some point we might want to make this do an auto-commit to actually rebuild the documentation, but this will be good enough for now.

**Review by Pierre-Sassoulas — APPROVED** [review]
👌

**Comment by DanielNoord** [comment]
Just tested and this works. Link on `readthedocs` links correctly and the one on the GitHub README as well. That's a pretty nice accomplishment (if I say so myself 😄).

**Pull request merged**